freqtrade/optimize/backslapping.py

```diff
@@ -0,0 +1,808 @@
+                """
+                # TODO :add handing here to record none closed open trades
+                
+
+                if debug:
+                    print(bslap_pair_results)
+                break
+            else:
+                """
+                Add trade to backtest looking results list of dicts
+                Loop back to look for more trades.
+                """
+
+                # We added +1 to t_exit_ind if the exit was a stop-loss, to not exit early in the IF of this ELSE
+                # removing the +1 here so prices match.
+                if t_exit_type == SellType.STOP_LOSS:
+                    t_exit_ind = t_exit_ind - 1
+
+                # Build trade dictionary
+                ## In general if a field can be calculated later from other fields leave blank here
+                ## Its X(number of trades faster) to calc all in a single vector than 1 trade at a time
+
+                # create a new dict
+                close_index: int = t_exit_ind
+                bslap_result = {}  # Must have at start or we end up with a list of multiple same last result
+                bslap_result["pair"] = pair
+                bslap_result["stoploss"] = stop
+                bslap_result["profit_percent"] = ""  # To be 1 vector calc across trades when loop complete
+                bslap_result["profit_abs"] = ""  # To be 1 vector calc across trades when loop complete
+                bslap_result["open_time"] = np_bslap_dates[t_open_ind + 1]  # use numpy array, pandas 20x slower
+                bslap_result["close_time"] = np_bslap_dates[close_index]  # use numpy array, pandas 20x slower
+                bslap_result["open_index"] = t_open_ind + 1  # +1 as we buy on next.
+                bslap_result["close_index"] = close_index
+                bslap_result["trade_duration"] = ""  # To be 1 vector calc across trades when loop complete
+                bslap_result["open_at_end"] = False
+                bslap_result["open_rate"] = round(np_trade_enter_price, 15)
+                bslap_result["close_rate"] = round(np_trade_exit_price, 15)
+                bslap_result["exit_type"] = t_exit_type
+                bslap_result["sell_reason"] = t_exit_type #duplicated, but I don't care
+                # append the dict to the list and print list
+                bslap_pair_results.append(bslap_result)
+
+                if t_exit_type is SellType.STOP_LOSS:
+                    stop_stops_count = stop_stops_count + 1
+
+                if debug:
+                    print("The trade dict is: \n", bslap_result)
+                    print("Trades dicts in list after append are: \n ", bslap_pair_results)
+
+                """
+                Loop back to start. t_exit_last becomes where loop
+                will seek to open new trades from.
+                Push index on 1 to not open on close
+                """
+                t_exit_last = t_exit_ind + 1
+
```

Comment:
Add test to break the loop if we're at the end of th DF here, after the  
 `t_exit_last = t_exit_ind + 1` +1 push on
```
'''
                Sense Check +1 is not past the end of the Dataframe, if so break
                '''
                if t_exit_ind >= len(bslap.index):
                    if debug:
                        print("t_exit_ind is the last row in the dataframe - breaking to next pair")
                    break
```

Reply:
Is it a question of performance? 
By the way i don't even use backslapping.py, I pasted the code in Edge. 
Also: I would like to rename "backslap" function with your permission. it is a source of confusion for the community. I would like to name it "detect_stops_and_sells". tell me what you think. the backslapping.py should be removed eventually